### PR TITLE
Implicit model with changes from master

### PR DESCRIPTION
**Pull request by ekallevik**

### CODE REVIEW

File: model/implicit_constraints.py [hunk 1]
```diff
@@ -89,16 +111,31 @@ def add_mapping_shift_to_demand(self, x, y):
                 quicksum(
                     quicksum(
                         x[e, t_marked, v] for t_marked in self.combined_time_periods
-                        if t - v + 1 <= t_marked <= t
+                        if t - v + self.time_step <= t_marked <= t
+                        if (e,t_marked,v) in x
+                        #
```
Comment (ekallevik) [inline model/implicit_constraints.py:116]: Should this be deleted @sondrenaerland ?

File: model/implicit_constraints.py [hunk 2]
```diff
@@ -89,16 +111,31 @@ def add_mapping_shift_to_demand(self, x, y):
...
+                        # and
+                        # t_marked + v - self.time_step in self.combined_time_periods
                     )
                     for v in self.shift_durations["work"]
-                )
-                == quicksum(y[c, e, t] for c in self.competencies if y.get((c, e, t)))
+                ) == quicksum(y[c, e, t] for c in self.competencies if y.get((c, e, t)))
                 for e in self.employees
                 for t in self.combined_time_periods
+                #for t in self.combined_time_periods if t < 165.75
```
Comment (ekallevik) [inline model/implicit_constraints.py:124]: Should this be deleted @sondrenaerland ?

File: model/implicit_constraints.py [hunk 3]
```diff
@@ -89,16 +111,31 @@ def add_mapping_shift_to_demand(self, x, y):
...
+        self.model.addConstrs(
+            (
+                x[e, 165.75, 2.25]
+                == quicksum(y[c, e, t] for c in self.competencies if y.get((c, e, t)))
+                for e in self.employees
+                for t in self.combined_time_periods if t >= 165.75
+            ),
+            name="mapping_shift_to_demand_special_case"
+        )
+        """
```
Comment (ekallevik) [inline model/implicit_constraints.py:138]: Should this be deleted @sondrenaerland ?

File: model/implicit_constraints.py [hunk 4]
```diff
@@ -47,6 +51,24 @@ def __init__(self, model, var, data):
...
+                        if t_marked + v - self.time_step in self.time_periods_with_no_demand
+                    )
+                    for v in self.shift_durations["work"]
+                ) == 0
+                for e in self.employees
+            ),
+            name="no_work_shift_ending_in_no_demand_allowed"
+        )
+    """
+
```
Comment (ekallevik) [inline model/implicit_constraints.py:71]: Should this be deleted @sondrenaerland ?

File: model/implicit_constraints.py [hunk 5]
```diff
@@ -31,6 +34,7 @@ def __init__(self, model, var, data):
         self.contracted_hours = data["staff"]["employee_contracted_hours"]
 
         # Adding constraints
+        #self.add_no_shift_while_no_demand(var.x)
```
Comment (sondrenaerland) [inline model/implicit_constraints.py:37]: Denne kan fjernes.

File: model/implicit_constraints.py (hunk 4 above)
Comment (sondrenaerland) [reply model/implicit_constraints.py:71]: Jess, den kan fjernes!

File: model/implicit_constraints.py (hunk 2 above)
Comment (sondrenaerland) [reply model/implicit_constraints.py:124]: Kan fjernes!

File: model/implicit_constraints.py (hunk 1 above)
Comment (sondrenaerland) [reply model/implicit_constraints.py:116]: Disse tre linjene kan fjernes ja!

File: model/implicit_constraints.py (hunk 3 above)
Comment (sondrenaerland) [reply model/implicit_constraints.py:138]: Fjern!

**Pull request merged**